_resources/port1.0/group/haskell_cabal-1.0.tcl

```diff
@@ -19,77 +19,118 @@
 # haskell_cabal.bin
 #   The cabal binary. Defaults to ${prefix}/bin/cabal.
 #
-# haskell_cabal.default_args
-#   Default arguments for cabal used across invocations in configure, build, and
-#   destroot phases. Defaults to
-#       --prefix=${prefix}
-#       --bindir=${prefix}/bin
-#       --libdir=${prefix}/lib
-#       --libsubdir=${name}
-#       --dynlibdir=${prefix}/lib
-#       --libexecdir=${prefix}/libexec
-#       --libexecsubdir=${name}
-#       --datadir=${prefix}/share/${name}
-#       --docdir=${prefix}/share/doc/${name}
-#       --htmldir=${prefix}/share/doc/${name}
-#       --sysconfdir=${prefix}/etc/${name}
-#       --enable-documentation
-#       --enable-relocatable
+# haskell_cabal.cabal_root
+#   The root path for cabal. Defaults to ${workpath}/.home/.cabal .
 
 proc haskell_cabal.add_dependencies {} {
     global name
     if { ${name} ne "cabal" } {
-        depends_build-append port:cabal \
-                             port:ghc
+        depends_build-append \
+            port:bash \
+            port:cabal \
+            port:ghc
     }
 }
 port::register_callback haskell_cabal.add_dependencies
 
+options haskell_cabal.cabal_root
+default haskell_cabal.cabal_root {${workpath}/.home/.cabal}
+
+post-extract {
+    xinstall -m 0755 -d [option haskell_cabal.cabal_root]
+    set cabal_config_fd [open "[option haskell_cabal.cabal_root]/config" w+]
+    set cabal_install_version [exec bash -c "[option haskell_cabal.bin] --version | grep -E -o -e '\[\[:digit:].]+' | head -1 || true"]
+    set cabal_library_version [exec bash -c "[option haskell_cabal.bin] --version | grep -E -o -e '\[\[:digit:].]+' | tail -1 || true"]
```

Comment:
Not a friend of the `port:bash` dependency for a task that (a) could probably be done with macOS' builtin `/bin/sh`, and (b) could be done with Tcl.

I'd write the code to do this with Tcl, but unfortunately stack doesn't build on my macOS 11, so I can't generate test output. Can you provide me with the output of `cabal --version` so I can suggest a Tcl alternative to this?

I figure it might be something like this:

```suggestion
    set cabal_version_lines [regexp -all -inline "\[0-9.\]+" [exec ${haskell_cabal.bin} --version]
    set cabal_install_version [lindex $cabal_versions 0]
    set cabal_library_version [lindex $cabal_versions end]
```

Reply:
Yes, thank you. Much smarter. Please see my version of `tcl`. I've verified both using `tclsh` and `port configure panddoc` using #10096.